module/dkg/client.go

```diff
@@ -183,13 +183,23 @@ func (c *Client) SubmitResult(groupPublicKey crypto.PublicKey, publicKeys []cryp
 	// we have done here. Group Public key first followed by the individual public keys
 	finalSubmission := make([]cadence.Value, 0, len(publicKeys))
 
-	// initially append group public key
-	trimmedGroupHexString := trim0x(groupPublicKey.String())
-	finalSubmission = append(finalSubmission, cadence.NewString(trimmedGroupHexString))
+	// first append group public key
+	if groupPublicKey != nil {
+		trimmedGroupHexString := trim0x(groupPublicKey.String())
+		finalSubmission = append(finalSubmission, cadence.NewString(trimmedGroupHexString))
+	} else {
+		finalSubmission = append(finalSubmission, cadence.String(""))
```

Comment:
We'll need to use optional types here.
```suggestion
		finalSubmission = append(finalSubmission, cadence.NewOptional(nil))
```